core/src/main/java/cucumber/runtime/LinePredicate.java

```diff
@@ -0,0 +1,31 @@
+package cucumber.runtime;
+
+import gherkin.events.PickleEvent;
+import gherkin.pickles.PickleLocation;
+
+import java.util.List;
+import java.util.Map;
+
+public class LinePredicate implements PicklePredicate {
+    private Map<String, List<Long>> lineFilters;
+
+    public LinePredicate(Map<String, List<Long>> lineFilters) {
+        this.lineFilters = lineFilters;
+    }
+
+    @Override
+    public boolean apply(PickleEvent pickleEvent) {
+        String picklePath = pickleEvent.uri;
+        if (!lineFilters.containsKey(picklePath)) {
+            return true;
```

Comment:
I think this should be false. If a Pickle isn't referenced in the rerun file it shouldn't be added to the list of steps to execute.